platform/linux-generic/arch/aarch64/odp_crypto_armv8.c

```diff
@@ -137,6 +154,280 @@ null_crypto_routine(odp_packet_t pkt ODP_UNUSED,
 	return ODP_CRYPTO_ALG_ERR_NONE;
 }
 
+static
+odp_crypto_alg_err_t aes_gcm_encrypt(odp_packet_t pkt,
+				     const odp_crypto_packet_op_param_t *param,
+				     odp_crypto_generic_session_t *session)
+{
+	armv8_cipher_state_t cs = {
+		.counter = {
+			.d = {0, 0}
+		}
+	};
+	armv8_cipher_constants_t cc = {
+		.mode = 0
+	};
+	uint8_t *iv_ptr;
+	uint64_t nonce_length, nonce_byte_length;
+	uint64_t plaintext_length, plaintext_byte_length;
+	uint64_t aad_length, aad_byte_length;
+	int ret = 0;
+
+	/* Clear the allocated buffers */
+	memset(global->input, 0, _ARM_CRYPTO_MAX_DATA_LENGTH);
+	memset(global->nonce, 0, _ARM_CRYPTO_MAX_IV_LENGTH);
+	memset(global->aad, 0, _ARM_CRYPTO_MAX_AAD_LENGTH);
+	memset(global->digest, 0, _ARM_CRYPTO_MAX_DIGEST_LENGTH);
+	memset(global->output, 0, (_ARM_CRYPTO_MAX_DATA_LENGTH +
+	       _ARM_CRYPTO_MAX_DIGEST_LENGTH));
+
+	nonce_length = (session->p.cipher_iv.length) * 8;
+	/* IV length must be a multiple of 16 */
+	nonce_byte_length = ((nonce_length + 127) & ~127UL) >> 3;
+
+	if (param->cipher_iv_ptr)
+		iv_ptr = param->cipher_iv_ptr;
+	else if (session->p.cipher_iv.data)
+		iv_ptr = session->cipher.iv_data;
+	else
+		return ODP_CRYPTO_ALG_ERR_IV_INVALID;
+
+	/* Copy nonce and pad it to nonce_byte_length */
+	for (uint64_t i = 0; i < session->p.cipher_iv.length; i++)
+		global->nonce[i] = iv_ptr[i];
```

Comment:
This could just be a memcpy with the constant known IV length.

Reply:
Okay, I will replace it. Thanks Janne

> This code crashes in crypto validation tests in my environment:
> 
> ```
> Suite: odp_crypto_sync_inp
>   Test: crypto_test_enc_alg_null ...passed
>   Test: crypto_test_dec_alg_null ...passed
>   Test: crypto_test_enc_alg_aes_gcm ...*** stack smashing detected ***: terminated
> Aborted (core dumped)
> ```
> 
> Did you try running make check with your crypto lib enabled?

Hi Janne, I have run make check with the crypto library enabled on my local environment and all tests passed. With the new job we've added to the CI, the validation tests also pass. Can you please tell me more about your environment so that we can narrow down the issue?

Reply:
I will make this change. Thanks!